src/osemgrep/cli_scan/Scan_subcommand.ml

```diff
@@ -715,7 +715,8 @@ let run_scan_conf (conf : Scan_CLI.conf) : Exit_code.t =
       ~registry_caching:conf.registry_caching conf.rules_source
   in
   let rules_and_origins =
-    Lwt_platform.run (Lwt.pick (rules_and_origins :: spinner_ls))
+    if !Common.jsoo then Lwt_platform.run rules_and_origins
```

Comment:
We shouldn't need this since `spinner_ls` should just be `[]` if we're jsoo.

Reply:
ack. I resorted to some genetic programming (peppering in `!Common.jsoo` even when it shouldn't matter) in order to try resolve the js failure and lsp hang

Reply:
We probably just need to leverage the virtual module setup so we can be sure we don't pull in the `Lwt_unix` stuff. Austin and I were talking a bit earlier. cc @ajbt200128

Reply:
It looks like we want `Lwt_unix.timeout n`, I'll include a platform agnostic version in a PR

Reply:
Think Austin and I resolved. @zzeleznick I'll probably put throw up a PR onto this branch maybe.

Reply:
amazing 😲 !